File: android/versioned-abis/expoview-abi39_0_0/src/main/java/abi39_0_0/expo/modules/battery/PowerSaverReceiver.java
package abi39_0_0.expo.modules.battery;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.PowerManager;

public class PowerSaverReceiver extends BroadcastReceiver {
  public static String INTENT_ACTION = "expo.modules.battery.PowerSaverReceiver";

  public PowerSaverReceiver() {
  }

  @Override
  public void onReceive(Context context, Intent intent) {
    PowerManager powerManager = (PowerManager) BatteryModule.mContext.getApplicationContext().getSystemService(Context.POWER_SERVICE);
    boolean isLowPowerMode = powerManager.isPowerSaveMode();

    BatteryModule.onLowPowerModeChange(isLowPowerMode);
  }
}

File: android/versioned-abis/expoview-abi49_0_0/src/main/java/abi49_0_0/expo/modules/battery/PowerSaverReceiver.kt
package abi49_0_0.expo.modules.battery

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.PowerManager
import abi49_0_0.expo.modules.core.interfaces.services.EventEmitter

private val POWERMODE_EVENT_NAME = "Expo.powerModeDidChange"

class PowerSaverReceiver(private val eventEmitter: EventEmitter?) : BroadcastReceiver() {
  private fun onLowPowerModeChange(lowPowerMode: Boolean) {
    eventEmitter?.emit(
      POWERMODE_EVENT_NAME,
      Bundle().apply {
        putBoolean("lowPowerMode", lowPowerMode)
      }
    )
  }

  override fun onReceive(context: Context, intent: Intent) {
    val powerManager = context.applicationContext.getSystemService(Context.POWER_SERVICE) as PowerManager
    val isLowPowerMode = powerManager.isPowerSaveMode
    onLowPowerModeChange(isLowPowerMode)
  }
}
